cwms-data-api/src/main/java/cwms/cda/data/dao/StreamDao.java

```diff
@@ -1,130 +1,257 @@
 
+    static final int STREAM_OFFICE_ID_COLUMN_INDEX = 1;
+    static final int STREAM_STREAM_ID_COLUMN_INDEX = 2;
+    static final int STATIONING_STARTS_DS_COLUMN_INDEX = 3;
+    static final int STREAM_FLOWS_INTO_STREAM_COLUMN_INDEX = 4;
+    static final int STREAM_FLOWS_INTO_STATION_COLUMN_INDEX = 5;
+    static final int STREAM_FLOWS_INTO_BANK_COLUMN_INDEX = 6;
+    static final int STREAM_DIVERTS_FROM_STREAM_COLUMN_INDEX = 7;
+    static final int STREAM_DIVERTS_FROM_STATION_COLUMN_INDEX = 8;
+    static final int STREAM_DIVERTS_FROM_BANK_COLUMN_INDEX = 9;
+    static final int STREAM_STREAM_LENGTH_COLUMN_INDEX = 10;
+    static final int STREAM_AVERAGE_SLOPE_COLUMN_INDEX = 11;
+    static final int STREAM_COMMENTS_COLUMN_INDEX = 12;
+    private static final String DB_STATION_UNITS = "km";
+    public static final String DB_STREAM_SLOPE_UNITS = "%";
 
-public class StreamDao extends JooqDao<Stream> {
     public StreamDao(DSLContext dsl) {
         super(dsl);
     }
 
-    public Stream getStream(String streamId, String unitSystem, String officeId) {
-        String pStationUnit = UnitSystem.EN.value().equals(unitSystem) ? Unit.MILE.getValue() :
-                Unit.KILOMETER.getValue();
-        CwmsDbStreamJooq streamJooq = new CwmsDbStreamJooq();
-
-        return connectionResult(dsl, c -> {
-            StreamT streamResult = streamJooq.retrieveStreamF(c, streamId, pStationUnit, officeId);
-            return new Stream.Builder(streamId, streamResult.getStartsDownstream(),
-                    streamResult.getLength(), streamResult.getOfficeId())
-                    .withDivertingStreamId(streamResult.getDivertsFromStream())
-                    .withDiversionStation(streamResult.getDivertsFromStation())
-                    .withDiversionBank(streamResult.getDivertsFromBank())
-                    .withReceivingStreamId(streamResult.getFlowsIntoStream())
-                    .withConfluenceStation(streamResult.getFlowsIntoStation())
-                    .withConfluenceBank(streamResult.getFlowsIntoBank())
-                    .withComment(streamResult.getComments())
-                    .withAverageSlope(streamResult.getAverageSlope())
-                    .withStreamLocations(getStreamLocationsOnStream(streamId, unitSystem, officeId))
-                    .withTributaries(getTributaries(streamId, unitSystem, officeId))
-                    .withStreamReaches(getReaches(streamId, officeId))
-                    .build();
+    /**
+     * Retrieve a list of streams
+     * @param officeIdMask - the office id mask
+     * @param streamIdMask - the stream id mask
+     * @return a list of streams
+     */
+    public List<Stream> retrieveStreams(String officeIdMask, String streamIdMask) {
+        return connectionResult(dsl, conn -> {
+            setOffice(conn, officeIdMask);
+            ResultSet streams = CWMS_STREAM_PACKAGE.call_CAT_STREAMS(DSL.using(conn).configuration(), streamIdMask,
```

Comment:
@MikeNeilson - do you want these retrievals to be changed to tables queries within CDA? (would start making paging doable) or are we waiting until we get the jOOQ codegen available with table read access for everyone?

Reply:
No specific preference, I'm generally in favor of whatever the easiest way to get the data we need is.

Though if I was listing a preference is would be (in order):
1. JOOQ
2. JDBC
3. hec-cwms-data-access DAOs (but really only becuase it keeps looking like an odd circular dependency to me.)

Reply:
Yeah I think we can avoid 2/3 is almost all cases.

For 1. it's just the decision of coding against the PL/SQL packages, jOOQ DSL with codegenned tables/views, or jOOQ DSL with hardcoded table/view names. For what Bryson is doing here, PL/SQL packages hit the easiest way to get data we need.